File: python/samples/getting_started_with_processes/step03/steps/external_step.py
# Copyright (c) Microsoft. All rights reserved.


from typing import Any

from semantic_kernel.functions import kernel_function
from semantic_kernel.processes.kernel_process import (
    KernelProcessEventVisibility,
    KernelProcessStep,
    KernelProcessStepContext,
)


class ExternalStep(KernelProcessStep):
    external_event_name: str

    def __init__(self, external_event_name: str):
        super().__init__(external_event_name=external_event_name)

    @kernel_function()
    async def emit_external_event(self, context: KernelProcessStepContext, data: Any):
        await context.emit_event(
            process_event=self.external_event_name, data=data, visibility=KernelProcessEventVisibility.Public
        )

File: dotnet/samples/GettingStartedWithProcesses/Step03/Steps/ExternalStep.cs
// Copyright (c) Microsoft. All rights reserved.

using Microsoft.SemanticKernel;

namespace Step03.Steps;

/// <summary>
/// Step used in the Processes Samples:
/// - Step_03_FoodPreparation.cs
/// </summary>
public class ExternalStep(string externalEventName) : KernelProcessStep
{
    private readonly string _externalEventName = externalEventName;

    [KernelFunction]
    public async Task EmitExternalEventAsync(KernelProcessStepContext context, object data)
    {
        await context.EmitEventAsync(this._externalEventName, data);
    }
}
